Fabric.Authorization.Domain/Services/UserService.cs

```diff
@@ -29,5 +45,38 @@ public UserService(IUserStore userStore)
         {
             return await _userStore.Exists($"{subjectId}:{identityProvider}");
         }
+
+        public async Task<User> AddRolesToUser(IList<Role> rolesToAdd, string subjectId, string identityProvider)
+        {
+            var user = await _userStore.Get($"{subjectId}:{identityProvider}");
+            var grainSecurableItems = rolesToAdd.Select(r => new Tuple<string, string>(r.Grain, r.SecurableItem))
+                .Distinct();
+            var existingRoles = new List<Role>();
+            foreach (var tuple in grainSecurableItems)
+            {
+                existingRoles.AddRange(await _roleStore.GetRoles(tuple.Item1, tuple.Item2));
+            }
+
+            var exceptions = new List<Exception>();
+            foreach (var role in rolesToAdd)
+            {
```

Comment:
Should we use the `role.Name` in the error messages below?

Reply:
We've overriden ToString on Role, so it should get written out w/ the {role} variable.

Reply:
Oh I didn't notice the `{role}`.